python/tvm/tir/schedule/schedule.py

```diff
@@ -2111,6 +2112,82 @@ def after_unannotate(a: T.handle, b: T.handle) -> None:
             self, block_or_loop, ann_key
         )
 
+    ########## Schedule: Layout transformation ##########
+
+    @type_checked
+    def transform_layout(
+        self,
+        block: BlockRV,
```

Comment:
Is there an advantage to passing in `block`, `buffer_index`, and `is_write_index` rather than passing in the buffer to be transformed?  Since all use of the buffer need to be updated, not just those within the specified `block`, it looks like these three parameters are only used to identify the buffer to be transformed.

Reply:
There is no `get_buffer` primitive in TIR, so users have to use `buffer_index + is_write_index` to specify the buffer. I agree `BufferType` enum will be helpful

Reply:
Does this relate to similar meta schedule constraints as well?  That is, a user could explicitly find the buffer object they want to update, but the meta schedule wouldn't be able to find the corresponding buffer in a similar schedule, and so the transformation wouldn't be tunable?

Reply:
Yes. User interact the meta schedule with random variables (`BlockRV` / `LoopRV`) that can have different values during tuning